File: z-clients/javascript/src/models/getMetricsOut.ts
// this file is @generated
import {
    type MetricOut,
    MetricOutSerializer,
} from './metricOut';

export interface GetMetricsOut {
    metrics: MetricOut[];
}

export const GetMetricsOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): GetMetricsOut {
        return {
            metrics: object['metrics'].map((item: MetricOut) => MetricOutSerializer._fromJsonObject(item)),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: GetMetricsOut): any {
        return {
            'metrics': self.metrics.map((item) => MetricOutSerializer._toJsonObject(item)),
        };
    }
}

File: z-clients/rust/src/models/get_metrics_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::metric_out::MetricOut;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GetMetricsOut {
    pub metrics: Vec<MetricOut>,
}

impl GetMetricsOut {
    pub fn new(metrics: Vec<MetricOut>) -> Self {
        Self { metrics }
    }
}
